_field-types/supported-field-types/derived.md

```diff
@@ -0,0 +1,903 @@
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response adds highlighting to the `derived_request_object.request` field:
+
+<details markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "took": 5,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 2,
+    "hits": [
+      {
+        "_index": "logs_object",
+        "_id": "1",
+        "_score": 2,
+        "_source": {
+          "request_object": """{"@timestamp": 894030400, "clientip":"61.177.2.0", "request": "GET /english/venues/images/venue_header.gif HTTP/1.0", "status": 200, "size": 711}"""
+        },
+        "fields": {
+          "derived_request_object.request": [
+            "GET /english/venues/images/venue_header.gif HTTP/1.0"
+          ],
+          "derived_request_object.clientip": [
+            "61.177.2.0"
+          ]
+        },
+        "highlight": {
+          "derived_request_object.request": [
+            "GET /english/venues/<em>images</em>/venue_header.gif HTTP/1.0"
+          ]
+        }
+      },
+      {
+        "_index": "logs_object",
+        "_id": "4",
+        "_score": 2,
+        "_source": {
+          "request_object": """{"@timestamp": 894340400, "clientip":"61.177.2.0", "request": "GET /english/images/venue_bu_city_on.gif HTTP/1.0", "status": 400, "size": 1397}
+"""
+        },
+        "fields": {
+          "derived_request_object.request": [
+            "GET /english/images/venue_bu_city_on.gif HTTP/1.0"
+          ],
+          "derived_request_object.clientip": [
+            "61.177.2.0"
+          ]
+        },
+        "highlight": {
+          "derived_request_object.request": [
+            "GET /english/<em>images</em>/venue_bu_city_on.gif HTTP/1.0"
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+</details>
+
+### Inferred subfield type 
+
+Type inference is based on the same logic as [Dynamic mapping]({{site.url}}{{site.baseurl}}/opensearch/mappings#dynamic-mapping). Instead of inferring the subfield type from the first document, a random sample of documents is used to infer the type. If the subfield isn't found in any documents from the random sample, type inference fails and logs a warning. For subfields that seldom occur in documents, consider defining the explicit field type. Using dynamic type inference for such subfields may result in a query returning no results, a behavior similar to that of a missing field. 
+
+### Explicit subfield type
+
+To define the explicit subfield type, provide the `type` parameter in the `properties` object. The following example defines an explicit type for the `derived_logs_object.is_active` field as `boolean`. Because this field is only present in one of the documents, its type inference might fail, so it's important to define the explicit type:
```

Comment:
```suggestion
To define the explicit subfield type, provide the `type` parameter in the `properties` object. The following example defines `boolean` as the explicit type for the `derived_logs_object.is_active` field. Because this field is only present in one of the documents, its type inference might fail, so it's important to define the explicit type:
```

Reply:
Reworded.